### PR TITLE
Add delete experiment endpoint

### PR DESCRIPTION
**Pull request by ManInFez**

### CODE REVIEW

File: ert_storage/endpoints/experiments.py [hunk 1]
```diff
@@ -93,3 +93,10 @@ async def get_experiment_metadata(
     """
     experiment = db.query(ds.Experiment).get(experiment_id)
     return experiment.metadata_dict
+
+
+@router.post("/experiments/{experiment_id}/delete")
```
Comment (dotfloat) [inline ert_storage/endpoints/experiments.py:98]:
```suggestion
@router.delete("/experiments/{experiment_id}")
```

You then use HTTP's `DELETE` verb.

File: tests/integration/test_experiments.py [hunk 2]
```diff
@@ -32,3 +35,43 @@ def test_ensembles(client, create_experiment, create_ensemble):
...
+            f"/experiments/{experiment_id}/observations",
+            json=dict(
+                name=name,
+                values=obs["values"],
+                errors=obs["errors"],
+                x_axis=obs["x_axis"],
+            ),
+        )
+    resp = client.post_check(f"/experiments/{experiment_id}/delete")
+    assert len(client.get_check(f"/experiments").json()) == 0
```
Comment (dotfloat) [inline tests/integration/test_experiments.py:77]: Also test that the observations that have been created have been deleted.

Comment (dotfloat) [reply tests/integration/test_experiments.py:77]: Ie, I would expect a 404 if I try to access the observations tied to the experiment.

File: ert_storage/endpoints/experiments.py (hunk 1 above)
Comment (ManInFez) [reply ert_storage/endpoints/experiments.py:98]: of course 🙈

File: tests/integration/test_experiments.py (hunk 2 above)
Comment (ManInFez) [reply tests/integration/test_experiments.py:81]: Done!

**Review by dotfloat — APPROVED** [review]
DWISOTT (Does what is says on the tin)

**Pull request merged**